### PR TITLE
[MIG]auth_oauth: Migrated in new API

### PR DESCRIPTION
**Pull request by atp-odoo**
Task: https://www.odoo.com/web#id=12142&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.LAtwWb5mX6XbW0WS

### CODE REVIEW

File: addons/auth_oauth/__openerp__.py [hunk 1]
```diff
@@ -28,17 +8,17 @@
...
-        'auth_oauth_data.yml',
         'auth_oauth_view.xml',
-        'security/ir.model.access.csv',
+        'auth_oauth_data.yml',
+        'res_users.xml',
         'res_config.xml',
+        'security/ir.model.access.csv',
         'views/auth_oauth_login.xml',
     ],
     'installable': True,
```
Comment (atp-odoo) [inline addons/auth_oauth/__openerp__.py]: remove  'auto_install': False, default value is false.

File: addons/auth_oauth/auth_oauth_data.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<openerp>
```
Comment (atp-odoo) [inline addons/auth_oauth/auth_oauth_data.xml]: add encoding="utf-8"

File: addons/auth_oauth/auth_oauth_view.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<openerp>
```
Comment (atp-odoo) [inline addons/auth_oauth/auth_oauth_view.xml]: Add encoding="utf-8"? and move in views directory

File: addons/auth_oauth/res_config.py [hunk 4]
```diff
@@ -1,67 +1,47 @@
...
         }
 
-    def set_oauth_providers(self, cr, uid, ids, context=None):
-        google_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'auth_oauth', 'provider_google')[1]
-        facebook_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'auth_oauth', 'provider_facebook')[1]
-        config = self.browse(cr, uid, ids[0], context=context)
+    @api.multi
+    def set_oauth_providers(self):
+        google_provider = self.env['ir.model.data'].xmlid_to_object('auth_oauth.provider_google')
+        facebook_provider = self.env['ir.model.data'].xmlid_to_object('auth_oauth.provider_facebook')
```
Comment (atp-odoo) [inline addons/auth_oauth/res_config.py]: 2 times self.env['ir.model.data'] assign in variable IrModelData

File: addons/auth_oauth/res_config.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<openerp>
```
Comment (atp-odoo) [inline addons/auth_oauth/res_config.xml]: Add encoding="utf-8" and move into views

File: addons/auth_oauth/res_users.py [hunk 6]
```diff
@@ -58,15 +59,14 @@ def _auth_oauth_signin(self, cr, uid, provider, validation, params, context=None
...
                 raise openerp.exceptions.AccessDenied()
             assert len(user_ids) == 1
-            user = self.browse(cr, uid, user_ids[0], context=context)
-            user.write({'oauth_access_token': params['access_token']})
-            return user.login
+            user_ids[0].write({'oauth_access_token': params['access_token']})
+            return user_ids[0].login
         except openerp.exceptions.AccessDenied, access_denied_exception:
-            if context and context.get('no_user_creation'):
+            if self._context.get('no_user_creation'):
```
Comment (atp-odoo) [inline addons/auth_oauth/res_users.py]: use self.env.context

File: addons/auth_oauth/res_users.py [hunk 7]
```diff
@@ -83,35 +83,35 @@ def _auth_oauth_signin(self, cr, uid, provider, validation, params, context=None
...
         if not validation.get('user_id'):
             raise openerp.exceptions.AccessDenied()
         # retrieve and sign in user
-        login = self._auth_oauth_signin(cr, uid, provider, validation, params, context=context)
+        login = self._auth_oauth_signin(provider, validation, params)
         if not login:
             raise openerp.exceptions.AccessDenied()
         # return user credentials
-        return (cr.dbname, login, access_token)
+        return (self._cr.dbname, login, access_token)
```
Comment (atp-odoo) [inline addons/auth_oauth/res_users.py]: Use self.env.cr.dbname

**Comment by atp-odoo** [comment]
1 ) don't reindent all the xml to minimize the diff.

**Comment by xmo-odoo** [comment]
> 1 ) don't reindent all the xml to minimize the diff.

Probably a good idea, but FWIW you can add `?w=1` to the URL to display whitespace-ignoring diffs.

File: addons/auth_oauth/controllers/main.py [hunk 8]
```diff
@@ -1,19 +1,17 @@
...
-from openerp import SUPERUSER_ID
-from openerp import http
-from openerp.http import request
-from openerp.addons.web.controllers.main import db_monodb, ensure_db, set_cookie_and_redirect, login_and_redirect
+from openerp import http, _
 from openerp.addons.auth_signup.controllers.main import AuthSignupHome as Home
-from openerp.modules.registry import RegistryManager
-from openerp.tools.translate import _
+from openerp.addons.web.controllers.main import db_monodb, ensure_db, set_cookie_and_redirect, login_and_redirect
+from openerp.http import request
```
Comment (atp-odoo) [inline addons/auth_oauth/controllers/main.py]: reorder import

File: addons/auth_oauth/controllers/main.py [hunk 9]
```diff
@@ -42,16 +40,17 @@ def wrapper(self, *a, **kw):
...
 # Controller
 #----------------------------------------------------------
 class OAuthLogin(Home):
     def list_providers(self):
         try:
-            provider_obj = request.registry.get('auth.oauth.provider')
-            providers = provider_obj.search_read(request.cr, SUPERUSER_ID, [('enabled', '=', True), ('auth_endpoint', '!=', False), ('validation_endpoint', '!=', False)])
-            # TODO in forwardport: remove conditions on 'auth_endpoint' and 'validation_endpoint' when these fields will be 'required' in model
+            providers = request.env['auth.oauth.provider'].sudo().search_read([('enabled', '=', True), 
+                                                                               ('auth_endpoint', '!=', False),
```
Comment (atp-odoo) [inline addons/auth_oauth/controllers/main.py]: why not search method instead of search_read ?

File: addons/auth_oauth/ir_configparameter.py [hunk 10]
```diff
@@ -1,14 +1,14 @@
...
 
+    #TODO: Need to migrate
     def init(self, cr, force=False):
-        super(ir_configparameter, self).init(cr, force=force)
+        super(IrConfigParameter, self).init(cr, force=force)
         if force:
-            IMD = self.pool['ir.model.data']
-            oauth_oe = IMD.xmlid_to_object(cr, SUPERUSER_ID, 'auth_oauth.provider_openerp')
-            dbuuid = self.get_param(cr, SUPERUSER_ID, 'database.uuid')
+            oauth_oe = self.env['ir.model.data'].sudo().xmlid_to_object('auth_oauth.provider_openerp')
```
Comment (atp-odoo) [inline addons/auth_oauth/ir_configparameter.py]: env.ref

File: addons/auth_oauth/res_config.py [hunk 11]
```diff
@@ -1,67 +1,48 @@
...
-    def get_oauth_providers(self, cr, uid, fields, context=None):
-        google_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'auth_oauth', 'provider_google')[1]
-        facebook_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'auth_oauth', 'provider_facebook')[1]
-        rg = self.pool.get('auth.oauth.provider').read(cr, uid, [google_id], ['enabled','client_id'], context=context)
-        rf = self.pool.get('auth.oauth.provider').read(cr, uid, [facebook_id], ['enabled','client_id'], context=context)
+    @api.model
+    def get_oauth_providers(self, fields):
+        IrModelData = self.env['ir.model.data']
+        google_provider = IrModelData.xmlid_to_object('auth_oauth.provider_google')
+        facebook_provider = IrModelData.xmlid_to_object('auth_oauth.provider_facebook')
```
Comment (atp-odoo) [inline addons/auth_oauth/res_config.py]: env.ref

File: addons/auth_oauth/controllers/main.py [hunk 12]
```diff
@@ -181,14 +178,11 @@ def oea(self, **kw):
...
-        registry = RegistryManager.get(dbname)
-        with registry.cursor() as cr:
-            IMD = registry['ir.model.data']
-            try:
-                model, provider_id = IMD.get_object_reference(cr, SUPERUSER_ID, 'auth_oauth', 'provider_openerp')
-            except ValueError:
-                return set_cookie_and_redirect('/web?db=%s' % dbname)
-            assert model == 'auth.oauth.provider'
+        try:
+            model, provider_id = request.env['ir.model.data'].xmlid_to_res_model_res_id('auth_oauth.provider_openerp')
```
Comment (atp-odoo) [inline addons/auth_oauth/controllers/main.py]: request.env.ref

File: addons/auth_oauth/ir_configparameter.py [hunk 13]
```diff
@@ -1,12 +1,12 @@
...
+
+class IrConfigParameter(models.Model):
     _inherit = 'ir.config_parameter'
 
     def init(self, cr, force=False):
-        super(ir_configparameter, self).init(cr, force=force)
+        super(IrConfigParameter, self).init(cr, force=force)
         if force:
             IMD = self.pool['ir.model.data']
             oauth_oe = IMD.xmlid_to_object(cr, SUPERUSER_ID, 'auth_oauth.provider_openerp')
```
Comment (atp-odoo) [inline addons/auth_oauth/ir_configparameter.py]: same here request.env.ref with superuser

File: addons/auth_oauth/auth_oauth.py [hunk 14]
```diff
@@ -1,25 +1,21 @@
-from openerp.osv import osv, fields
```
Comment (rga-odoo) [inline addons/auth_oauth/auth_oauth.py]: move it to model direcotry

File: addons/auth_oauth/auth_oauth.py [hunk 15]
```diff
@@ -1,25 +1,21 @@
...
+# -*- coding: utf-8 -*-
 
-class auth_oauth_provider(osv.osv):
+from openerp import api, fields, models
```
Comment (rga-odoo) [inline addons/auth_oauth/auth_oauth.py]: unused import api

File: addons/auth_oauth/controllers/main.py [hunk 16]
```diff
@@ -1,19 +1,17 @@
+# -*- coding: utf-8 -*-
+
 import functools
 import logging
-
 import simplejson
 import urlparse
```
Comment (rga-odoo) [inline addons/auth_oauth/controllers/main.py]: unused import urlparse

File: addons/auth_oauth/ir_configparameter.py [hunk 17]
```diff
@@ -1,14 +1,13 @@
...
+class IrConfigParameter(models.Model):
     _inherit = 'ir.config_parameter'
 
     def init(self, cr, force=False):
-        super(ir_configparameter, self).init(cr, force=force)
+        super(IrConfigParameter, self).init(cr, force=force)
         if force:
-            IMD = self.pool['ir.model.data']
-            oauth_oe = IMD.xmlid_to_object(cr, SUPERUSER_ID, 'auth_oauth.provider_openerp')
+            oauth_oe = self.env(user=SUPERUSER_ID).ref('auth_oauth.provider_openerp')
```
Comment (rga-odoo) [inline addons/auth_oauth/ir_configparameter.py]: It's broken. we could not get env here

File: addons/auth_oauth/res_config.py [hunk 18]
```diff
@@ -1,67 +1,47 @@
...
-        facebook_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'auth_oauth', 'provider_facebook')[1]
-        config = self.browse(cr, uid, ids[0], context=context)
+    @api.multi
+    def set_oauth_providers(self):
+        google_provider = self.env.ref('auth_oauth.provider_google')
+        facebook_provider = self.env.ref('auth_oauth.provider_facebook')
         rg = {
-            'enabled':config.auth_oauth_google_enabled,
-            'client_id':config.auth_oauth_google_client_id,
+            'enabled':self.auth_oauth_google_enabled,
```
Comment (rga-odoo) [inline addons/auth_oauth/res_config.py]: lint your code,  E231 missing whitespace after ':'

File: addons/auth_oauth/res_config.py [hunk 19]
```diff
@@ -1,67 +1,47 @@
...
-#
-##############################################################################
-
-from openerp.osv import osv, fields
 
 import logging
+
+from openerp import api, fields, models
+
 _logger = logging.getLogger(__name__)
```
Comment (rga-odoo) [inline addons/auth_oauth/res_config.py]: unused _logger

**Comment by rgo-odoo** [comment]
- Remove the commented code from auth_oauth_data.xml
- place 'id' as the first attribute in views
- Rename,
    ir_configparameter.py ==> ir_config_parameter.py
    auth_oauth_login.xml ==> auth_oauth_templates.xml
    auth_oauth_view.xml ==> auth_oauth_views.xml

**Pull request closed without merging**